src/packs/checker.rs

```diff
@@ -262,9 +263,40 @@ pub(crate) fn update(
     Ok(())
 }
 
+pub(crate) fn remove_unnecessary_dependencies(
+    configuration: &Configuration,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let unnecessary_dependencies = get_unnecessary_dependencies(configuration);
+    for (package_name, dependency_name) in unnecessary_dependencies.iter() {
```

Comment:
Writing the pack to disk with sorted, de-duped dependencies is actually done for us!

All you need to do is:
1. Create a new pack from the old one with the new set of dependencies
2. Call `write_pack_to_disk`

I added a commit on `main`, b232cb6, to demonstrate.